### DISCUSSION TITLE
How to use a compound function with debounce?

### DISCUSSION THREAD

**Discussion by wsw70** [quasarframework/quasar#10357] [General - Components / Directives / etc]
I use debounce to save dynamic-but-irregular data (think: someone typing):

    watch(allNotes, debounce(writeAllNotes, 3000), {deep: true})

This works great.

I wanted to expand the function a tiny bit by also setting a flag (whihc is reset in `writeAllNotes`):

    watch(allNotes, () => { notesNotSaved.value = true ; debounce(writeAllNotes, 3000) }, {deep: true})

This does not work that well anymore:
- `notesNotSaved` is set to `true` as expected
- the debounce function does not work anymore (it is not called after 3 seconds)

The workaround I found is to split `watch` but I would like to understand what I do not do correctly:

```
watch(allNotes, debounce(writeAllNotes, 3000), {deep: true})
watch(allNotes, () => {notesNotSaved.value = true}, {deep: true})
```

**Comment by pdanpdan** [comment]
Assign `debounce(writeAllNotes, 3000)` to a variable and call that function in the watch function
